src/site/content/en/blog/multi-screen-window-placement/index.md

```diff
@@ -168,106 +171,154 @@ The new `window-placement` permission can be queried with the
-to each other, on top of each other, etc.). There is also data now for each screen to show whether
-it is an `internal` one and whether it is a `primary` one.
-Note that the built-in screen
+Information about the connected screens is available in the `screens` array. Note how the value of
+`left` for the iPad starts at `1680`, which is exactly the `width` of the built-in display. This
+allows me to determine exactly how the screens are arranged logically (next to each other, on top of
+each other, etc.). There is also data now for each screen to show whether it is an `isInternal` one
+and whether it is an `isPrimary` one. Note that the built-in screen
 [is not necessarily the primary screen](https://osxdaily.com/2010/04/27/set-the-primary-display-mac/#:~:text=Click%20on%20the%20Display%20icon,primary%20display%20for%20your%20Mac).
-Both also have an `id`, which, if
-persisted across browser sessions, allows for window arrangements to be restored.
+Both also have a `label` and an `id`, which, if persisted across browser sessions, allows for window
+arrangements to be restored. Information about the possible input methods are available in the
+`pointerTypes` array.
+
+The field `currentScreen` is a live object corresponding to the current `window.screen`. The object
+is updated on cross-screen window placements or device changes.
 
 ### The `screenschange` event
 
 The only thing missing now is a way to detect when my screen setup changes. A new event,
-`screenschange`, does exactly that: it fires whenever the screen constellation
-is modified.
-(Notice that "screens" is plural in the event name.)
-It also fires when the resolution of one of the connected screens changes or when a
-new or an existing screen is (physically or virtually in the case of Sidecar) plugged in or unplugged.
+`screenschange`, does exactly that: it fires whenever the screen constellation is modified. (Notice
+that "screens" is plural in the event name.) It also fires when the resolution of one of the
+connected screens changes or when a new or an existing screen is (physically or virtually in the
+case of Sidecar) plugged in or unplugged.
 
 Note that you need to look up the new screen details asynchronously, the `screenschange` event
-itself does not provide this data. This may [change in the
-future](https://github.com/webscreens/window-placement/issues/28). For now you
-can look up the screen details by calling `window.getScreens()` as shown below.
+itself does not provide this data. This may
+[change in the future](https://github.com/webscreens/window-placement/issues/28). For now you can
+look up the screen details by calling `window.getScreens()` as shown below.
 
 ```js
-window.addEventListener('screenschange', async (event) => {
-  console.log('I am there, but mostly useless', event);
+const screensInterface = await window.getScreens();
+screensInterface.addEventListener('screenschange', async (event) => {
   const details = await window.getScreens();
```

Comment:
Consider avoiding the repeated call to getScreens. One practical example here might be something like:

const screensInterface = await window.getScreens();
let cachedScreensLength = screensInterface.screens.length;
screensInterface.addEventListener('screenschange', event => {
  if (screensInterface.screens.length != cachedScreensLength) {
    console.log('The screen count changed from' + cachedScreensLength + ' to' screensInterface.screens.length);
    cachedScreensLength = screensInterface.screens.length;
  }
});

Generally, `screenschange` should correspond to a change in the length of `screens`, but technically it's possible that a single event might fire if N screens were added and N screens were removed simultaneously...

Reply:
Noted and changed.